examples/sdk_resampling_example.py

```diff
@@ -0,0 +1,142 @@
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+
+import asyncio
+import random
+from datetime import datetime
+from typing import Dict
+
+import pytz
+from frequenz.channels import Broadcast, Receiver, Select, Sender
+
+from frequenz.sdk.data_ingestion.resampling.data_provider import DataProvider
+from frequenz.sdk.data_ingestion.resampling.subscriptions import (
+    Subscription,
+    SubscriptionRequest,
+    SubscriptionResponse,
+)
+from frequenz.sdk.data_ingestion.resampling.timeseries_resampler import Sample
+from frequenz.sdk.data_ingestion.resampling.timeseries_resampling_actor import (
+    TimeSeriesResamplingActor,
+)
+
+
+async def data_sender(sender: Sender[Sample]) -> None:
+    """Send a message to the channel every 0.2 seconds.
+
+    Args:
+        sender: channel to be used to send random component data
+    """
+    while True:
+        value = random.uniform(1.0, 100.0)
+        await sender.send(Sample(timestamp=datetime.now(tz=pytz.UTC), value=value))
+        await asyncio.sleep(random.uniform(0.1, 0.5))
+
+
+class DummyDataProvider(DataProvider):
+    """Dummy Data Provider for presentation purposes."""
+
+    def __init__(self) -> None:
+        """Create an instance and initialize receivers."""
+        self._receivers: Dict[Subscription, Receiver[Sample]] = {}
+
+    async def subscribe(self, subscription: Subscription) -> SubscriptionResponse:
+        """Pass a subscription to the data source and return a subscription response.
+
+        Args:
+            subscription: object of the subscription (time_series_id, start_time)
+
+        Returns:
+            response containing a sample receiver for the time series id provided in
+            the subscription object
+        """
+        receiver = self._receivers.get(subscription)
+        if receiver is None:
+            channel = Broadcast[Sample](f"Source {subscription.time_series_id}")
+            asyncio.create_task(data_sender(channel.get_sender()))
+            receiver = channel.get_receiver()
+            self._receivers[subscription] = receiver
+        return SubscriptionResponse(receiver=receiver, subscription=subscription)
+
+
+async def run() -> None:
+    """Run main functions that initializes and creates everything."""
+    data_provider: DataProvider = DummyDataProvider()
+
+    # Create a channel for receiving subscription responses from the resampling actor
+    subscription_response_channel = Broadcast[SubscriptionResponse](
+        "Subscription Response Channel"
+    )
+    subscription_response_sender = subscription_response_channel.get_sender()
+    subscription_response_receiver = subscription_response_channel.get_receiver()
+
+    # Create a channel for sending subscription requests to the resampling actor
+    subscription_request_channel = Broadcast[SubscriptionRequest](
+        "Subscription Request Channel"
+    )
+    subscription_request_sender = subscription_request_channel.get_sender()
```

Comment:
Makes sense, I didn't think about that. Clients could store what they requested and only accept responses if the underlying `subscription` is what they requested, but your solution is better, because responses shouldn't be sent out to all the subscribers. 

I'll wait for further instructions on how to change the subscriptions.